modules/flowable-engine/src/test/java/org/flowable/engine/test/api/runtime/ProcessInstanceIdentityLinksTest.java

```diff
@@ -67,49 +67,33 @@ public void testCandidateGroupLink() {
-        IdentityLink identityLink = identityLinks.get(0);
-
-        assertEquals("muppets", identityLink.getGroupId());
-        assertNull("kermit", identityLink.getUserId());
-        assertEquals(IdentityLinkType.PARTICIPANT, identityLink.getType());
-        assertEquals(processInstanceId, identityLink.getProcessInstanceId());
-
-        assertEquals(1, identityLinks.size());
+        assertThat(identityLinks)
+                .extracting(IdentityLink::getGroupId, IdentityLink::getUserId, IdentityLink::getType, IdentityLink::getProcessInstanceId)
+                .containsExactly(tuple("muppets", null, IdentityLinkType.PARTICIPANT, processInstanceId));
 
         if (HistoryTestHelper.isHistoryLevelAtLeast(HistoryLevel.AUDIT, processEngineConfiguration)) {
             List<Event> processInstanceEvents = runtimeService.getProcessInstanceEvents(processInstanceId);
-            assertEquals(1, processInstanceEvents.size());
-            Event processInstanceEvent = processInstanceEvents.get(0);
-            assertEquals(Event.ACTION_ADD_GROUP_LINK, processInstanceEvent.getAction());
-            List<String> processInstanceEventMessageParts = processInstanceEvent.getMessageParts();
-            assertEquals("muppets", processInstanceEventMessageParts.get(0));
-            assertEquals(IdentityLinkType.PARTICIPANT, processInstanceEventMessageParts.get(1));
-            assertEquals(2, processInstanceEventMessageParts.size());
+            assertThat(processInstanceEvents)
+                    .extracting(Event::getAction)
+                    .containsExactly(Event.ACTION_ADD_GROUP_LINK);
+            List<String> processInstanceEventMessageParts = processInstanceEvents.get(0).getMessageParts();
+            assertThat(processInstanceEventMessageParts)
+                    .containsOnly("muppets", IdentityLinkType.PARTICIPANT);
         }
 
         runtimeService.deleteParticipantGroup(processInstanceId, "muppets");
 
         if (HistoryTestHelper.isHistoryLevelAtLeast(HistoryLevel.AUDIT, processEngineConfiguration)) {
             List<Event> processInstanceEvents = runtimeService.getProcessInstanceEvents(processInstanceId);
-            Event processIsntanceEvent = findProcessInstanceEvent(processInstanceEvents, Event.ACTION_DELETE_GROUP_LINK);
-            assertEquals(Event.ACTION_DELETE_GROUP_LINK, processIsntanceEvent.getAction());
-            List<String> processInstanceEventMessageParts = processIsntanceEvent.getMessageParts();
-            assertEquals("muppets", processInstanceEventMessageParts.get(0));
-            assertEquals(IdentityLinkType.PARTICIPANT, processInstanceEventMessageParts.get(1));
-            assertEquals(2, processInstanceEventMessageParts.size());
-            assertEquals(2, processInstanceEvents.size());
+            assertThat(processInstanceEvents)
+                    .extracting(Event::getAction)
+                    .contains(Event.ACTION_DELETE_GROUP_LINK);
```

Comment:
Why not `containsExactly(Event.ACTION_DELETE_GROUP_LINK, Event.ACTION_DELETE_GROUP_LINK)` this way the size of 2 is also verified. Or add `hasSize(2)`